File: pricing-service-java/src/main/java/com/example/retail/javaapp/cache/QuoteComputation.java
package com.example.retail.javaapp.cache;

import com.example.retail.contract.AppliedPromotion;
import com.example.retail.contract.FulfillmentOption;

import java.math.BigDecimal;
import java.util.List;

public record QuoteComputation(
        BigDecimal basePrice,
        List<AppliedPromotion> appliedPromotions,
        BigDecimal finalPrice,
        List<FulfillmentOption> fulfillmentOptions
) {
}

File: pricing-service-kotlin/src/main/kotlin/com/example/retail/kotlinapp/cache/QuoteComputation.kt
package com.example.retail.kotlinapp.cache

import com.example.retail.contract.AppliedPromotion
import com.example.retail.contract.FulfillmentOption
import java.math.BigDecimal

data class QuoteComputation(
    val basePrice: BigDecimal,
    val appliedPromotions: List<AppliedPromotion>,
    val finalPrice: BigDecimal,
    val fulfillmentOptions: List<FulfillmentOption>,
)

